controllers/nstemplateset/nstemplatetier_test.go

```diff
@@ -234,6 +255,34 @@ func TestProcessWithTTRTable(t *testing.T) {
 			runtimeParams: map[string]string{}, // SPACE_NAME missing
 			expectedError: `map has no entry for key "SPACE_NAME"`,
 		},
+
+		{
+			name:          "completely invalid YAML in step 1",
+			templates:     []string{completelyInvalidYAML},
+			staticParams:  []toolchainv1alpha1.Parameter{},
+			runtimeParams: standardRuntimeParams,
+			expectedError: "yaml: line 2: mapping values are not allowed in this context",
+		},
+		{
+			name:      "GVK-specific filtering like cluster resources",
+			templates: []string{namespaceTemplate, configMapTemplate, clusterResourceQuotaTemplate},
+			staticParams: []toolchainv1alpha1.Parameter{
+				{Name: "NAMESPACE", Value: "test-ns"},
+				{Name: "CONFIG_VALUE", Value: "test-config"},
+			},
+			runtimeParams: standardRuntimeParams,
+			filters: []template.FilterFunc{retainObjectsOfSameGVK(schema.GroupVersionKind{
+				Group:   "quota.openshift.io",
+				Version: "v1",
+				Kind:    "ClusterResourceQuota",
+			})},
```

Comment:
wouldn't it be easier to rely on the already existing filter `template.RetainNamespaces`? The filtering related to cluster-scoped objects will go away soon